alphagov/paas-bootstrap
# [#137364635] Add GPG signature verification

## What

We intend to push the create/destroy pipelines to the Concourses created by paas-bootstrap (although only the 'create' pipeline to persistent environtments) and eventually have these pipelines auto-triggered, therefore we want to have an extra guard in place to make sure the change has been authorised by a member of the team.

The implementation was copied from paas-cf[1]
 
[1] https://github.com/alphagov/paas-cf/tree/811da924b62da9de28e6bd0d7d7c385f768982ef

## How to review

Review in conjunction with https://github.com/alphagov/paas-team-manual/pull/124

First of all a code review. I haven't added the signature verification to the concourse-lite-self-terminate.yml pipeline, but in my opinion it doesn't need to.

Check the Travis run as it runs the unit tests for the pull request merging script lifted from paas-cf. You can use the script to merge this pull request if you want to, or test it on a dummy PR.

To test it fails to import the paas-bootstrap resource when a commit hasn't been signed:

* deploy this branch:

  ```
  BRANCH=gpg-signature-verification make dev deployer-concourse bootstrap
  ```

* trigger the create pipeline manually. The paas-bootstrap resource should fail with:

  ```
  The commit <hash> is not signed
  ```

To test it successfully imports the repository when the commit has been signed:

* create a copy of this branch.
* merge a change into it and sign the merge:

  ```
  git merge -S --no-ff some-other-branch
  ```

* push up your branch and deploy from it:

  ```
  BRANCH=your-branch-name make dev deployer-concourse bootstrap
  ```

* trigger the create pipeline manually. Pause the second job to stop it actually completing. The paas-bootstrap resource should successfully be imported into the first job.
* delete the remote version of your branch 

### ⚠️ Before merging

Remove the temporary commit and merge using the new `make` target. It is up to you whether you want to test the `make` target separately, but it is a lift-and-shift from paas-cf.

## Who can review

Anyone but me
